src/PostgREST/Request/DbRequestBuilder.hs

```diff
@@ -342,6 +345,7 @@ mutateRequest mutation schema tName ApiRequest{..} pkCols readReq = mapLeft ApiR
     rootOrder = maybe [] snd $ find (\(x, _) -> null x) qsOrder
     combinedLogic = foldr addFilterToLogicForest logic qsFiltersRoot
     body = payRaw <$> iPayload -- the body is assumed to be json at this stage(ApiRequest validates)
+    noBodyFilterLimit = null qsColumns && null qsLogic && null qsFilters && null qsRanges && (isNothing body || S.null iColumns)
```

Comment:
This check would be better at the `ApiRequest` level since it doesn't depend on the db. This way we avoid taking a pool connection and opening an empty transaction when this error happens.